pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCuda.hpp

```diff
@@ -252,6 +252,31 @@ class TNCuda : public TNCudaBase<PrecisionT, Derived> {
                                                         host_data_size);
     }
 
+    /**
+     * @brief Check if the provided MPS has the correct dimension for C++
+     * backend.
+     *
+     * @param MPS_shape_source Dimension list of incoming MPS.
+     */
+
+    void MPSShapeCheck(
+        const std::vector<std::vector<std::size_t>> &MPS_shape_source) {
+        bool sameShape = sitesExtents_ == MPS_shape_source;
+        if (!sameShape) {
```

Comment:
can we just combine these to `if (sitesExtents != MPS_shape_source)`

Reply:
Done!